internal/management/controller/roles/runnable.go

```diff
@@ -177,40 +256,73 @@ func synchronizeRoles(
 		roleInSpecNamed[r.Name] = r
 	}
 
-	// 1. do any of the roles in the DB require update/delete?
+	rolesByAction := make(map[roleAction][]apiv1.RoleConfiguration)
+	// 1. find the next actions for the roles in the DB
 	roleInDBNamed := make(map[string]apiv1.RoleConfiguration)
 	for _, role := range rolesInDB {
 		roleInDBNamed[role.Name] = role
-		inSpec, found := roleInSpecNamed[role.Name]
+		inSpec, isInSpec := roleInSpecNamed[role.Name]
 		switch {
-		case found && inSpec.Ensure == apiv1.EnsureAbsent:
-			contextLog.Info("role in DB and Spec, but spec wants it absent. Deleting", "role", role.Name)
-			err = roleManager.Delete(ctx, role)
-			if err != nil {
-				return wrapErr(err)
-			}
-		case found && !areEquivalent(inSpec, role):
-			contextLog.Info("role in DB and Spec, are different. Updating", "role", role.Name)
-			err = roleManager.Update(ctx, inSpec)
-			if err != nil {
-				return wrapErr(err)
-			}
-		case !found:
-			contextLog.Debug("role in DB but not Spec. Ignoring it", "role", role.Name)
+		case ReservedRoles[role.Name]:
+			rolesByAction[roleIsReserved] = append(rolesByAction[roleIsReserved], role)
+		case isInSpec && inSpec.Ensure == apiv1.EnsureAbsent:
+			rolesByAction[roleDelete] = append(rolesByAction[roleDelete], role)
+		case isInSpec && !areEquivalent(inSpec, role):
```

Comment:
TODO need adopt to !areEquivalent or commentsUpdate or passwordNeedSync
wait for sync with cnp-3437, or we introduced in RoleCommentsUpdate ??
 and RolePasswordUpdate ?? action